### PR TITLE
fix: error from API response

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

(A few sentences about this PR)

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens/constants.go`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `supertokens/constants.go > version variable`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: recipe/passwordless/passwordlessfunctions.go [hunk 1]
```diff
@@ -30,6 +30,32 @@ import (
 var PasswordlessLoginEmailSentForTest bool = false
 var PasswordlessLoginSmsSentForTest bool = false
 
+func logAndReturnError(resp *http.Response, err error) error {
```
Comment (rishabhpoddar) [inline recipe/passwordless/passwordlessfunctions.go:33]: How are these changes tested?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: recipe/passwordless/passwordlessfunctions.go (hunk 1 above)
Comment (sattvikc) [reply recipe/passwordless/passwordlessfunctions.go:33]:
```
func main() {
	configValue := supertokens.TypeInput{
		Supertokens: &supertokens.ConnectionInfo{
			ConnectionURI: "http://localhost:3567",
			APIKey:        "someKey",
		},
		AppInfo: supertokens.AppInfo{
			AppName:       "SuperTokens",
			WebsiteDomain: "localhost:3000",
			APIDomain:     "localhost:8000",
		},
		RecipeList: []supertokens.Recipe{
			passwordless.Init(plessmodels.TypeInput{
				FlowType: "USER_INPUT_CODE",
				ContactMethodEmail: plessmodels.ContactMethodEmailConfig{
					Enabled: true,
				},
			}),
		},
	}
	err := supertokens.Init(configValue)
	if err != nil {
		panic(err)
	}

	mux := http.NewServeMux()
	testServer := httptest.NewServer(supertokens.Middleware(mux))
	defer testServer.Close()

	unittesting.PasswordlessEmailLoginRequest("sattvik@gmail.com", testServer.URL)
}
```

Through debugging the above code. Similar for SMS as well.

Comment (sattvikc) [reply recipe/passwordless/passwordlessfunctions.go:33]: image

Debug logs received

Comment (sattvikc) [reply recipe/passwordless/passwordlessfunctions.go:33]: image

Returned error

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**